mysql/datadog_checks/mysql/statements.py

```diff
@@ -172,15 +172,38 @@ def _query_summary_per_statement(self):
         rows = []  # type: List[PyMysqlRow]
 
         try:
-            with closing(self._get_db_connection().cursor(pymysql.cursors.DictCursor)) as cursor:
-                cursor.execute(sql_statement_summary)
-
-                rows = cursor.fetchall() or []  # type: ignore
+            rows = self._execute_query(sql_statement_summary)
         except (pymysql.err.InternalError, pymysql.err.OperationalError) as e:
+            self._check.count(
+                "dd.mysql.statements.error",
+                1,
+                tags=self._tags + ["error:{}".format(type(e))] + self._check._get_debug_tags(),
+                hostname=self._check.resolved_hostname,
+            )
             self.log.warning("Statement summary metrics are unavailable at this time: %s", e)
 
         return rows
 
+    def _execute_query(self, query):
+        rows = []  # type: List[PyMysqlRow]
+        try:
+            with closing(self._get_db_connection().cursor(pymysql.cursors.DictCursor)) as cursor:
+                cursor.execute(query)
+                rows = cursor.fetchall() or []  # type: ignore
+        # InterfaceError occurs when the physical connection is closed (e.g. a database restart).
+        # To recover, close the job's connection and let a new one get lazily recreated on the next run
+        except pymysql.err.InterfaceError as e:
```

Comment:
Should we just let these exceptions bubble up to the top? We already have some standardized code that will track exceptions and also do a shutdown of the job so the whole thing is recreated automatically on the next run. 

https://github.com/DataDog/integrations-core/blob/e2c6696e9ee84f27052737e14a657014ebc06e06/datadog_checks_base/datadog_checks/base/utils/db/utils.py#L245-L269 

I think this code is from before we moved the collection into a separate job so we wanted to ensure it didn't crash the main check.

Reply:
Done and tested. Had to remove the `catch` blocks from both `_collect_per_statement_metrics` and `collect_per_statement_metrics` but, with that, we get a clean job shutdown and restart and query metrics collection recovers cleanly.